specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json

```diff
@@ -3811,6 +4562,130 @@
         }
       }
     },
+    "StaticSiteDatabaseConnectionConfigurationFileOverview": {
+      "description": "A database connection configuration file",
+      "type": "object",
+      "properties": {
+        "fileName": {
+          "description": "The name of the configuration file.",
+          "type": "string",
+          "readOnly": true
+        },
+        "contents": {
+          "description": "The Base64 encoding of the file contents.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The type of configuration file.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-client-flatten": true
+    },
+    "DatabaseConnection": {
+      "description": "Static Site Database Connection ARM resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "DatabaseConnection resource specific properties",
+          "type": "object",
+          "properties": {
+            "resourceId": {
+              "description": "The resource id of the database.",
+              "type": "string"
+            },
+            "connectionIdentity": {
+              "description": "If present, the identity is used in conjunction with ConnectionString to connect to the database. Use of the system-assigned managed identity is indicated with the string 'SystemAssigned', while use of a user-assigned managed identity is indicated with the resource id of the managed identity resource.",
+              "type": "string"
+            },
+            "connectionString": {
+              "description": "The connection string to use to connect to the database.",
+              "type": "string",
+              "x-ms-secret": true
```

Comment:
>secret [](http://example.com/codeflow?start=20&length=6)

+do we also need x-ms-mutability: "create, write" or is it implied by x-ms-secret? I forget what the tooling needs here.